crypto/evp/ctrl_params_translate.c

```diff
@@ -1512,8 +1512,11 @@ static int get_payload_group_name(enum state state,
         return 0;
     }
 
-    if (ctx->p2 != NULL)
-        ctx->p1 = strlen(ctx->p2);
+    if (ctx->p2 == NULL) {
+        ERR_raise(ERR_LIB_EVP, EVP_R_UNSUPPORTED_KEY_TYPE);
+        return 0;
```

Comment:
I'm not sure why the original code was wrong? In any case this seems to make it an error for there to be no group name, where it wasn't an error before? Why? "Unsupported key type" doesn't seem like an appropriate reason code for there being no group name.

Reply:
ctx->p2 could still be NULL if either of the branches of the switch got a `NID_undef` result.  I assumed that that was an error condition and factored the code out of the switch.

Reply:
Hmm. It could be considered a success to get an empty group name for parameters that don't have a group name?

Reply:
I guess what is important is that we remain consistent with previous behaviour.

Reply:
It doesn't look like this is a 1.1.1 ctrl.  The only reference to this function doesn't have a ctrl and has a comment:

>      * The following contain no ctrls, they are exclusively here to extract
>      * key payloads from legacy keys, using OSSL_PARAMs, and rely entirely
>      * on |fixup_args| to pass the actual data.  The |fixup_args| should
>      * expect to get the EVP_PKEY pointer through |ctx->p2|.

@levitte, your thoughts here?

Reply:
So this is so that doing a "get" for a param on a legacy key works? So, it seems that the behaviour here should be consistent with what we do in the providers if we want to get a group name.

At least for DH/DSA it looks like we completely ignore an attempt to get a group name where no such group name exists for provided implementations:

https://github.com/openssl/openssl/blob/50864bd2f7bcbc76cb68c46c4f4c6ce57c683a5f/crypto/ffc/ffc_params.c#L271-L280

Reply:
Okay, I've switched to silently ignoring the request in the NULL case.